### ISSUE TITLE
Investigate trio loop sanity

### ISSUE THREAD

**Issue by vxgmichel** [Scille/parsec-cloud#557]
We need a tool to measure the amount of time each task run before before giving the control back to the trio loop. We want to make sure to get some warnings when this time never exceeds a certain limit (say 100 ms like [asyncio debug mode](https://docs.python.org/3/library/asyncio-dev.html#debug-mode)) as this would be the sign of incorrect IO management (or heavy CPU task).

Trio reference: https://trio.readthedocs.io/en/latest/reference-hazmat.html?highlight=instrument#debugging-and-instrumentation

**Comment by vxgmichel** [comment]
My findings:
```
2019-07-03 11:05:21 [warning  ] The previous step for task `parsec.core.sync_monitor.monitor_sync` took 1.195 s
Resuming afer:
  File "/home/vinmic/parsec-cloud/parsec/core/sync_monitor.py", line 156, in monitor_sync
    await sync_monitor.run(task_status=task_status)
```
And
```
2019-07-03 11:04:43 [warning  ] The last step for task `__main__.amain.<locals>.claim_task` took 0.092 s [parsec.core.monitoring]
```

**Comment by vxgmichel** [comment]
It turns out the `merge_buffers` function is very slow, even for low number of buffers. Consider the following test:
```python
def test_merge_buffers_performance():
    blocks = []
    dirty_size = 4 * 1024
    clean_size = 512 * 1024
    total_size = 2 * 1024 * 1024
    for offset in range(0, total_size, dirty_size):
        block = Buffer(offset, offset + dirty_size, None)
        blocks.append(block)

    merged = merge_buffers_with_limits_and_alignment(blocks, 0, total_size, clean_size)

    assert merged.start == 0
    assert merged.end == merged.size == total_size
    assert len(merged.spaces) == 4
    assert sum(len(space.buffers) for space in merged.spaces) == 512
    buffers = [buff.buffer for space in merged.spaces for buff in space.buffers]
    assert buffers == blocks
```

Here the merging of 512 buffers takes half a second (note that size does not matter here as we are not dealing with the data itself). Running this test with `pytest-profiling` shows that:
- 99 % of the time is spent sorting buffers using `sorted`
- 96 % of the time is spent comparing the items to sort (`__lt__` method)
- 84 % of the time is spent converting the buffer objects to tuple in order to make the comparison (using `attrs_to_tuple`)

**Comment by vxgmichel** [comment]
The `merge_buffer` performance issue is fixed in commit 3e2ec1a.
